### ISSUE TITLE
bug in /opt/tpot/bin/rules.sh when configuring IP in docker-compose.yml ports definition

### ISSUE THREAD

**Issue by relka** [telekom-security/tpotce#1060]
Hello,

When specifying an IP address in docker-compose.yml ports section, this results in a bug in rule.sh script. Let say I want to force a container to be reachable only from specific host IP 192.168.122.1:

```
....
ports:
     - "192.168.122.1:22:22"
```

This will result in rule.sh trying to setup iptable rule with port "192.168.122.1".

Simple solution is to retrieve all defined IPs on TPOT host, and to remove them from ports list (instead of just filtering 127.0.0.1).
This can be done by updating the fuGETPORTS  function in rule.sh in following way:

```
function fuGETPORTS {
### Get ports from docker-compose.yml

### filter all potentially included IPs in docker-compose.yml, and not only 127.0.0.1
myLOCALIPS=$(ifconfig | grep -Eo 'inet (addr:)?([0-9]*\.){3}[0-9]*' | grep -Eo '([0-9]*\.){3}[0-9]*')
while IFS= read -r line; 
	do IPSCLEANUP="${IPSCLEANUP} s/${line}// ;"
done <<< "$myLOCALIPS"

myDOCKERCOMPOSEPORTS=$(cat $myDOCKERCOMPOSEYML | yq -r '.services[].ports' | grep ':' | sed -e "$IPSCLEANUP" | tr -d '", ' | sed -e s/^:// | cut -f1 -d ':' )

myDOCKERCOMPOSEPORTS+=" $myHOSTPORTS"
myRULESPORTS=$(for i in $myDOCKERCOMPOSEPORTS; do echo $i; done | sort -gu)
echo "Setting up / removing these ports:"
echo "$myRULESPORTS"
}
```

BTW, we miss an option in tpot.yml files ( /opt/tpot/etc/compose/* ) to configure such IP globally, and to prevent docker-proxy to listen on ":::* " by default. It would be cool to add a global variable like ${TPOT_LISTENING_IP} (or whatever), and to prefix all ports definitions with that:

```
...
ports:
     - "${TPOT_LISTENING_IP}:22:22"
     - "${TPOT_LISTENING_IP}:8080:8080"
...
```

Impressive work anyway !

**Comment by t3chn0m4g3** [comment]
@relka Thanks. Can you elaborate on your use-case? From what I am understanding it seems you want to multi-home T-Pot with multiple IFs homed to different subnets while using the `docker-compose.yml` to achieve some sort of access control.

**Comment by relka** [comment]
Yes, that's exactly it. In our case, TPOT is hosted on a network where the administration is done through a dedicated physical LAN, and our need is to partition the flows so that the honeypots only listen from the production interface.

**Comment by relka** [comment]
just in case, in order to use the ' "${TPOT_LISTENING_IP}:22:22"' syntax in tpot.yml, I've slightly updated rules.sh and tpot.service files:

on rules.sh:
```
function fuGETPORTS {
...
myLOCALIPS=$(ifconfig | grep -Eo 'inet (addr:)?([0-9]*\.){3}[0-9]*' | grep -Eo '([0-9]*\.){3}[0-9]*')
IPSCLEANUP='s/\${TPOT_LISTENING_IP}//'
while IFS= read -r line; 
	do IPSCLEANUP="${IPSCLEANUP} ; s/${line}//"
done <<< "$myLOCALIPS"
...
```

and on tpot.service:
```
[Service]
Environment="TPOT_LISTENING_IP=192.168.122.1"
...
```

It may be alos simpler to use the *myLOCALIP* variable instead of  TPOT_LISTENING_IP for that - for example by exporting myLOCALIP to env in updateip.sh ?

**Comment by t3chn0m4g3** [comment]
For your use case this seems important, however after reviewing the pros and cons it does not seem the majority of the T-Pot community will profit from this. It also comes with possible security drawbacks as multi-homing is something not recommended for a honeypot.
Thank you for understanding.
